packages/react/src/components/Icon/Icon.tsx

```diff
@@ -45,50 +67,97 @@ export interface IconProps extends UIComponentProps, ColorComponentProps {
   xSpacing?: IconXSpacing
 }
 
-class Icon extends UIComponent<WithAsProp<IconProps>, any> {
-  static create: ShorthandFactory<IconProps>
+const Icon: React.FC<WithAsProp<IconProps>> & FluentComponentStaticProps = props => {
+  const {
+    accessibility,
+    alt,
+    'aria-label': ariaLabel,
+    bordered,
+    circular,
+    className,
+    color,
+    disabled,
+    design,
+    name,
+    outline,
+    rotate,
+    size,
+    styles,
+    variables,
+    xSpacing,
+  } = props
+
+  const context: ProviderContextPrepared = React.useContext(ThemeContext)
+  const { setStart, setEnd } = useTelemetry(Icon.displayName, context.telemetry)
+
+  setStart()
+
+  const getA11Props = useAccessibility(accessibility, {
+    debugName: Icon.displayName,
+    mapPropsToBehavior: () => ({
+      alt,
+      'aria-label': ariaLabel,
+    }),
+    rtl: context.rtl,
+  })
+  const { classes } = useStyles(Icon.displayName, {
+    className: Icon.className,
+    mapPropsToStyles: () => ({
+      bordered,
+      circular,
+      color,
+      disabled,
+      name,
+      outline,
+      rotate,
+      size,
+      xSpacing,
+    }),
+    mapPropsToInlineStyles: () => ({ className, design, styles, variables }),
+    rtl: context.rtl,
+  })
 
-  static className = 'ui-icon'
+  const ElementType = getElementType(props)
+  const unhandledProps = getUnhandledProps(Icon.handledProps, props)
 
-  static displayName = 'Icon'
+  const { icons = {} } = context.theme
+  const maybeIcon = icons[name]
+  const isSvgIcon = maybeIcon && maybeIcon.isSvg
```

Comment:
This can be sent as a prop in the mapPropsToStyles and simplify the logic in the styles. BTW, I don't see any updates in the iconStyles, please check those too.

Reply:
Good catch, did updates there